tpm.c

```diff
@@ -212,6 +251,12 @@ static EFI_STATUS tpm_log_event_raw(EFI_PHYSICAL_ADDRESS buf, UINTN size,
 EFI_STATUS tpm_log_event(EFI_PHYSICAL_ADDRESS buf, UINTN size, UINT8 pcr,
 			 const CHAR8 *description)
 {
+	EFI_STATUS efi_status;
+	efi_status = cc_log_event_raw(buf, size, pcr, description,
+				      strlen(description) + 1, EV_IPL, NULL);
+	if (EFI_ERROR(efi_status))
```

Comment:
Hmm, so `cc_log_event_raw()` is never called with hash != NULL.

Looks like we are missing something here, I guess the `tpm_log_event() ` callers should pass down the information whenever they log a PE binary or config data?

Reply:
(after looking at the complete tpm.c file).  There are three places calling `tpm_log_event_raw()`.  So, yes, either call `cc_log_event_raw()` from `tpm_log_event_raw()`, which indeed looks easiest, or add a `cc_log_event_raw()` call to all places calling `tpm_log_event_raw()`.

Reply:
Oh, I see where the "try again without PE_COFF flag" logic comes from, `tpm_log_event_raw()` does that too.
Would be interesting to know why the shim tpm code does that, it still looks wrong to me.

Reply:
> Oh, I see where the "try again without PE_COFF flag" logic comes from, `tpm_log_event_raw()` does that too. Would be interesting to know why the shim tpm code does that, it still looks wrong to me.

Unlike CC measurement via Intel TDX or AMD SEV, TPM measurement is done by hardware device, so I think some of hardwares may exist bug to support PE_COFF. that is why fallback code.